rktl_sim/nodes/simulator

```diff
@@ -358,5 +449,244 @@ class Simulator(object):
+        if car_name in self.car_effort_subs:
+            self.car_effort_subs[car_name].unregister()
+            del self.car_effort_subs[car_name]
+        if car_name in self.car_cmd_subs:
+            self.car_cmd_subs[car_name].unregister()
+            del self.car_cmd_subs[car_name]
+        # delete all of the cars publishers
+        if car_name in self.car_odom_pubs:
+            self.car_odom_pubs[car_name].unregister()
+            del self.car_odom_pubs[car_name]
+        if car_name in self.car_pose_pubs:
+            self.car_pose_pubs[car_name].unregister()
+            del self.car_pose_pubs[car_name]
+    # reseting vars to be safe
+    self.car_ids = {}
+    self.car_pose_pubs = {}
+    self.car_odom_pubs = {}
+    self.car_effort_subs = {}
+    self.car_cmd_subs = {}
+    self.reset_lock.release()
+
+    return DeleteCarResponse(True)
+
+
+def generate_all_cars(self):
+    car_configs = rospy.get_param('~cars', [])
+
+    for car_config in car_configs:
+        init_pose = None
+        if 'init_pose' in car_config:
+            init_pose = car_config['init_pose']
+
+        if 'name' not in car_config:
+            rospy.signal_shutdown('no "name" set for car config in sim')
+        car_name = car_config['name']
+        if 'randomize_pose' in car_config:
+            init_pose = None
+
+        self.car_ids[car_name] = self.sim.create_car(
+            'car', init_pose=init_pose, noise=self.car_noise, car_props=self.car_properties)
+
+        car_id = self.car_ids[car_name]
+        self.car_effort_subs[car_name] = rospy.Subscriber(
+            f'/cars/{car_name}/effort', ControlEffort,
+            self.effort_cb, callback_args=car_id)
+        self.car_cmd_subs[car_name] = rospy.Subscriber(
+            f'/cars/{car_name}/command', ControlCommand,
+            self.cmd_cb, callback_args=car_id)
+
+        if self.mode == SimulatorMode.REALISTIC:
+            self.car_pose_pubs[car_name] = rospy.Publisher(
+                f'/cars/{car_name}/pose_sync_early',
+                PoseWithCovarianceStamped, queue_size=1)
+            self.car_odom_pubs[car_name] = rospy.Publisher(
+                f'/cars/{car_name}/odom_truth', Odometry, queue_size=1)
+        elif self.mode == SimulatorMode.IDEAL:
+            self.car_odom_pubs[car_name] = rospy.Publisher(
+                f'/cars/{car_name}/odom', Odometry, queue_size=1)
+
+    return CreateCarResponse(True)
+
+
+def set_sensor_noise(self):
+    """
+    adds the sensor noise for the car and for the ball
+    @return: the sensor noises for the ball and for the car
+    """
+    sensor_noise = rospy.get_sim_param('~sensor_noise', None)
+    ball_noise = None
+    car_noise = None
+    if sensor_noise is not None and self.mode == SimulatorMode.REALISTIC:
+        ball_noise = sensor_noise.get('ball', None)
+        car_noise = sensor_noise.get('car', None)
+    self.car_noise = car_noise
+    self.ball_noise = ball_noise
+    return car_noise, ball_noise
```

Comment:
We should either set member variables or return the variable, doing both is a little overkill

Reply:
done!